controllers/drplacementcontrol_controller_test.go

```diff
@@ -1658,101 +1645,38 @@ func deleteDRPolicySync() {
+	} else {
+		latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateFenced
 	}
 
-	Eventually(func() bool {
-		latestDRCluster := getLatestDRCluster(cluster)
-
-		drClusterFencedCondition := getDRClusterCondition(&latestDRCluster.Status,
-			rmn.DRClusterConditionTypeFenced)
-		if drClusterFencedCondition == nil {
-			return false
-		}
-
-		if drClusterFencedCondition.ObservedGeneration != latestDRCluster.Generation {
-			return false
-		}
-
-		return drClusterFencedCondition.Status == metav1.ConditionTrue
-	}, timeout, interval).Should(BeTrue(), "failed to update DRCluster on time")
+	latestDRCluster = updateDRClusterParameters(latestDRCluster)
+	drclusterConditionExpectEventually(latestDRCluster, false, metav1.ConditionTrue,
+		Equal(controllers.DRClusterConditionReasonFenced), Ignore(),
+		rmn.DRClusterConditionTypeFenced)
 }
 
 func unfenceCluster(cluster string, manual bool) {
-	localRetries := 0
-	for localRetries < updateRetries {
-		latestDRCluster := getLatestDRCluster(cluster)
-		if manual {
-			latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateManuallyUnfenced
-		} else {
-			latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateUnfenced
-		}
-
-		err := k8sClient.Update(context.TODO(), latestDRCluster)
-		if errors.IsConflict(err) {
-			localRetries++
-
-			continue
-		}
-
-		Expect(err).NotTo(HaveOccurred())
-
-		break
+	latestDRCluster := getLatestDRCluster(cluster)
+	if manual {
+		latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateManuallyUnfenced
+	} else {
+		latestDRCluster.Spec.ClusterFence = rmn.ClusterFenceStateUnfenced
 	}
 
-	Eventually(func() bool {
-		latestDRCluster := getLatestDRCluster(cluster)
-
-		drClusterFencedCondition := getDRClusterCondition(&latestDRCluster.Status,
-			rmn.DRClusterConditionTypeFenced)
-		if drClusterFencedCondition == nil {
-			return false
-		}
-
-		if drClusterFencedCondition.ObservedGeneration != latestDRCluster.Generation {
-			return false
-		}
-
-		return drClusterFencedCondition.Status == metav1.ConditionFalse
-	}, timeout, interval).Should(BeTrue(), "failed to update DRCluster on time")
+	latestDRCluster = updateDRClusterParameters(latestDRCluster)
+	drclusterConditionExpectEventually(latestDRCluster, false, metav1.ConditionFalse,
+		BeElementOf(controllers.DRClusterConditionReasonUnfenced, controllers.DRClusterConditionReasonCleaning,
+			controllers.DRClusterConditionReasonClean),
+		Ignore(), rmn.DRClusterConditionTypeFenced)
 }
 
 func resetdrCluster(cluster string) {
-	localRetries := 0
-
-	for localRetries < updateRetries {
-		latestDRCluster := getLatestDRCluster(cluster)
-		latestDRCluster.Spec.ClusterFence = ""
-
-		err := k8sClient.Update(context.TODO(), latestDRCluster)
-		if errors.IsConflict(err) {
-			localRetries++
-
-			continue
-		}
-
-		Expect(err).NotTo(HaveOccurred())
-
-		break
-	}
+	latestDRCluster := getLatestDRCluster(cluster)
+	latestDRCluster.Spec.ClusterFence = ""
+	updateDRClusterParameters(latestDRCluster)
```

Comment:
Looks like good change for its own commit - other changes in this file are not related to this change, right?

Reply:
This commit description is: "Use updateDRClusterParameters in functions fenceCluster, unfenceCluster and resetdrCluster". I supposed, that if the action is the same - replace code with using helper - I must not create the separate commit for every function. If you want to separate - we need 3 different commits here

Reply:
Looks good now when we have the same change.